python/ray/tests/test_metrics_agent.py

```diff
@@ -252,42 +264,56 @@ def ping(self):
         prom_addresses.append(f"{addr}:{metrics_export_port}")
 
     # Make sure we can ping Prometheus endpoints.
-    def get_component_information(prom_addresses):
-        # TODO(sang): Add a core worker & gcs_server after adding metrics.
+    def fetch_prometheus(prom_addresses):
         components_dict = {}
+        metric_names = set()
         for address in prom_addresses:
             if address not in components_dict:
                 components_dict[address] = set()
             try:
                 response = requests.get(
                     "http://localhost:{}".format(metrics_export_port))
             except requests.exceptions.ConnectionError:
-                return components_dict
+                return components_dict, metric_names
 
             for line in response.text.split("\n"):
                 for family in text_string_to_metric_families(line):
                     for sample in family.samples:
                         # print(sample)
+                        metric_names.add(sample.name)
                         if "Component" in sample.labels:
                             components_dict[address].add(
                                 sample.labels["Component"])
-        return components_dict
+        return components_dict, metric_names
 
     def test_prometheus_endpoint():
         # TODO(sang): Add a core worker & gcs_server after adding metrics.
-        components_dict = get_component_information(prom_addresses)
-        COMPONENTS_CANDIDATES = {"raylet"}
-        return all(
-            COMPONENTS_CANDIDATES.issubset(components)
+        components_dict, metric_names = fetch_prometheus(prom_addresses)
+
+        # Raylet and GCS should be on every node
```

Comment:
Maybe mistake on your comment? GCS is only in the head node, and your code only contains raylet.